docs/modules/ROOT/pages/reference/extensions/core.adoc

```diff
@@ -104,6 +104,27 @@ camel.beans.customBeanWithSetterInjection.counter = 123
 
 As such, the class `PropertiesCustomBeanWithSetterInjection` needs to be link:https://quarkus.io/guides/writing-native-applications-tips#registering-for-reflection[registered for reflection], note that field access could be omitted in this case.
 
+==== Serialization in native mode
+
+If serialization support is requested via `quarkus.camel.native.reflection.serialization-enabled`, following classes are registered for serialization:
+
+* `java.lang.Boolean`
+* `java.lang.Byte`
+* `java.lang.Character`
+* `java.lang.Float`
+* `java.lang.Double`
+* `java.lang.Integer`
+* `java.lang.Long`
+* `java.lang.Number`
+* `java.lang.String`
+* `java.math.BigInteger`
+* `java.util.Date`
+* `java.util.HashMap`
+* `java.util.LinkedHashMap`
+* `org.apache.camel.support.DefaultExchangeHolder`
+
+Developers can register other classes for serialization via the `ReflectiveClassBuildItem`.
```

Comment:
`ReflectiveClassBuildItem` is for extension developers and this is rather a part of user guide. I think we should better omit this sentence or mention some way how application devs can register additional classes for serialization. Don't we have a property for that? (Or I wonder whether `@RegisterForReflection` shouldn't be modify to allow that too?)

Wouldn't [native-mode.adoc](https://camel.apache.org/camel-quarkus/latest/user-guide/native-mode.html) be a better place for this whole section about Serialization in native mode?

Reply:
yes, you are right . Native-mode.adoc seems to be better place to write about it and I'll try to omit the buildItem and add link to the native registration via properties.

Reply:
done